### DISCUSSION TITLE
'%s' tests on bonus 'space' flag

### DISCUSSION THREAD

**Discussion by denisgodoy** [Tripouille/printfTester#7] [Q&A] [answered]
Hi! I'm currently doing the project (new pdf) and using the tester, but I've faced a problem with bonus part2. According to man BSD, the 'space' flag has no effect on 's' conversions, however the tester does expect them. I'm doing bonus2 only, as the new pdf allows me to manage it partially, so I will not implement field width, '0', '.' and '-'. I'd like to ask if the tester is not correctly updated or is it related to Moulinette and the '% 1s' conversion?

**Comment by Chinimala** [comment]
Hello,

For the s conversion, can you post a link to that man, or send a screen of it ? On every man I look at, the space flag has no conversion specified (example : http://transit.iut2.upmf-grenoble.fr/cgi-bin/man/man2html?3+printf : "(a space) A blank should be left before a positive number (or empty string) produced by a signed conversion.").

For the mix with '0', '.' and '-' flags, yes the tester is not totally updated and won't be, so you'll have to manage with it.

Good luck !

**Reply by denisgodoy** [reply]
@Chinimala, according to printf(3) in https://www.freebsd.org/cgi/man.cgi?printf(3) and the mac environment we use for correction:
_` ' (space)  A blank should be left before a positive number produced by a signed conversion (a, A, d, e, E, f,	F, g, G, or i)._

Maybe this could lead to a false KO on the tester when 'make space' is used and bonus part 1 is not done (for field width management)? I haven't submitted the project yet, neither could find who did bonus part 2 only to effectively now what is expected by Moulinette.

![Captura de tela de 2021-09-17 09-54-04](https://user-images.githubusercontent.com/56933400/133787794-b06e495e-813e-49fb-8fd4-791355580ce4.png)

Thanks for replying.

**Comment by Chinimala** [comment, marked as answer]
You are right :) The space flag didn't have much sense anyway with a string. You can do a pull request to remove the useless tests if you want, or just ignore them. But you will still have false KOs on bonus flags + space width management, the tester is not maintained anymore.

**Reply by denisgodoy** [reply]
Thanks for the quick reply. I'll open a PR as soon as I submit the project to make sure everything is correct.
